Clean up custom console output with utils

cassaundra opened the pull request:
Use utility functions to simplify styling and make the existing styling and indentation more consistent.

A few instances of manual styling remain:
- In the deprecation messages in `set_version.rs` and `upgrade.rs` (these could use an aesthetic pass too though, since they look out of place).
- In the feature activation list of `set_version`.
- In cargo add, per review discussion.

This also serves as a step towards #682, as these utilities functions are similar in form to Cargo's shell output.

epage commented on src/util.rs:
````diff
@@ -8,3 +13,64 @@ pub fn colorize_stderr() -> ColorChoice {
+/// }
+/// ```
+pub fn print_warning(message: &str) -> CargoResult<()> {
+    print_colored_message("Warning:", message, Color::Yellow)
````
It should be `warning` and not `Warning` to match cargo

cassaundra commented:
Okay! I wasn't originally planning on directly mimicking Cargo's internal implementation of the shell utilities, but seeing the merits of that, I renamed the functions as requested. I also added a `justified` parameter, and applied that to `shell_warn`.

Per your request, I also reverted all changes made to cargo add. It turns out this is the only instance of an uppercase message anyway, inside of cargo add, for what it's worth: https://github.com/killercup/cargo-edit/blob/fdb12eb9f2957902e9f97f0ee5045e22405f2e94/src/bin/add/add.rs#L582-L585

cassaundra commented on src/bin/add/add.rs:
```diff
@@ -618,7 +618,7 @@ fn exec(mut args: AddArgs) -> CargoResult<()> {
     if was_sorted {
         if let Some(table) = manifest
-            .get_or_insert_table_mut(&args.get_section())
+            .get_table_mut(&args.get_section())
```
Ah, that's my bad. I reverted changes to the wrong version (pre-#711). I'll fix that.

cassaundra commented on src/bin/add/add.rs:
```diff
@@ -566,7 +566,7 @@ fn exec(mut args: AddArgs) -> CargoResult<()> {
 
             if !unknown_features.is_empty() {
-                shell_warn(&format!("Unrecognized features: {:?}", unknown_features))?;
+                shell_warn(&format!("unrecognized features: {:?}", unknown_features))?;
```
Okay! I feel plenty comfortable rewriting my commit history. I intentionally included that change in that commit in case you changed your mind about cargo add, but I definitely agree that it'd be better to rewrite those changes to appear more coherent outside the context of this PR.

epage replied:
I'd like to leave `src/bin/add` left along as much as possible.  If it was to be updated as part of another API changing, that is understandable but any improvements are a dead end.

cassaundra commented:
Okay! I've cleaned up the git history quite a bit. Should be good to merge.

epage commented:
Thanks!
